src/lib/homestore_backend/hs_shard_manager.cpp

```diff
@@ -313,7 +316,8 @@ void HSHomeObject::on_shard_message_commit(int64_t lsn, sisl::blob const& h, hom
     switch (header->msg_type) {
     case ReplicationMessageType::CREATE_SHARD_MSG: {
         auto sb = r_cast< shard_info_superblk const* >(h.cbytes() + sizeof(ReplicationMessageHeader));
-        auto const shard_info = sb->info;
+        auto shard_info = sb->info;
+        shard_info.lsn = lsn;
```

Comment:
should we also set lsn in line 354 for seal_shard?

Reply:
lsn is for create shard. To know when this shard was created so that we can ignore if it was created after snapshot taken.